scripts/create_sample_pcluster_config.sh

```diff
@@ -0,0 +1,122 @@
+export HEAD_SG=$(echo $OOD_STACK | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="HeadNodeSecurityGroup") | .OutputValue')
+export HEAD_POLICY=$(echo $OOD_STACK | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="HeadNodeIAMPolicyArn") | .OutputValue')
+export COMPUTE_SG=$(echo $OOD_STACK | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="ComputeNodeSecurityGroup") | .OutputValue')
+export COMPUTE_POLICY=$(echo $OOD_STACK | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="ComputeNodeIAMPolicyArn") | .OutputValue')
+export BUCKET_NAME=$(echo $OOD_STACK | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="ClusterConfigBucket") | .OutputValue')
+export LDAP_ENDPOINT=$(echo $OOD_STACK | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="LDAPNLBEndPoint") | .OutputValue')
+
+
+cat << EOF > ../pcluster-config.yml 
+HeadNode:
+  InstanceType: c5.large
+  Ssh:
+    KeyName: $SSH_KEY
+  Networking:
+    SubnetId: $SUBNET
+    AdditionalSecurityGroups:
+      - $HEAD_SG
+  LocalStorage:
+    RootVolume:
+      VolumeType: gp3
+      Size: 50
+  Iam:
+    AdditionalIamPolicies:
+      - Policy: arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+      - Policy: arn:aws:iam::aws:policy/AmazonS3FullAccess
+      - Policy: $HEAD_POLICY
+  CustomActions:
+    OnNodeConfigured:
+      Script: >-
+        s3://$BUCKET_NAME/pcluster_head_node.sh
+      Args:
+        - $STACK_NAME
+Scheduling:
+  Scheduler: slurm
+  SlurmQueues:
+    - Name: general
+      AllocationStrategy: lowest-price
+      ComputeResources:
+        - Name: general-cr
+          Instances:
+            - InstanceType: c5n.large
+          MinCount: 0
+          MaxCount: 4
+      Networking:
+        SubnetIds:
+          - $SUBNET
+        AdditionalSecurityGroups:
+          - $COMPUTE_SG
+      ComputeSettings:
+        LocalStorage:
+          RootVolume:
+            VolumeType: gp3
+            Size: 50
+      CustomActions:
+        OnNodeConfigured:
+          Script: >-
+            s3://$BUCKET_NAME/pcluster_worker_node.sh
+          Args:
+            - $STACK_NAME
+      Iam:
+        AdditionalIamPolicies:
+          - Policy: arn:aws:iam::aws:policy/AmazonS3FullAccess
+          - Policy: >-
+              $COMPUTE_POLICY
+    - Name: desktop
+      AllocationStrategy: lowest-price
+      ComputeResources:
+        - Name: desktop-cr
+          Instances:
+            - InstanceType: c5n.2xlarge
+          MinCount: 0
+          MaxCount: 10
+      Networking:
+        SubnetIds:
+          - $SUBNET
+        AdditionalSecurityGroups:
+          - $COMPUTE_SG
+      ComputeSettings:
+        LocalStorage:
+          RootVolume:
+            VolumeType: gp3
+            Size: 50
+      CustomActions:
+        OnNodeConfigured:
+          Script: >-
+            s3://$BUCKET_NAME/pcluster_worker_node_desktop.sh
+          Args:
+            - $STACK_NAME
+      Iam:
+        AdditionalIamPolicies:
+          - Policy: arn:aws:iam::aws:policy/AmazonS3FullAccess
```

Comment:
Can you help me understand the need for AmazonS3FullAccess? I'm concerned it doesn't align with principal of least privilege.

Reply:
removed.